android/src/main/java/com/alinz/parkerdan/shareextension/ShareModule.java

```diff
@@ -5,73 +5,85 @@
 import com.facebook.react.bridge.Promise;
 import com.facebook.react.bridge.ReactMethod;
 import com.facebook.react.bridge.WritableMap;
+import com.facebook.react.bridge.WritableArray;
 import com.facebook.react.bridge.Arguments;
 
 import android.app.Activity;
 import android.content.Intent;
 import android.net.Uri;
-
 import android.graphics.Bitmap;
-import java.io.InputStream;
 
+import java.io.InputStream;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.ArrayList;
 
 public class ShareModule extends ReactContextBaseJavaModule {
 
 
-  public ShareModule(ReactApplicationContext reactContext) {
-      super(reactContext);
-  }
-
-  @Override
-  public String getName() {
-      return "ReactNativeShareExtension";
-  }
-
-  @ReactMethod
-  public void close() {
-    getCurrentActivity().finish();
-  }
-
-  @ReactMethod
-  public void data(Promise promise) {
-      promise.resolve(processIntent());
-  }
-
-  public WritableMap processIntent() {
-      WritableMap map = Arguments.createMap();
-
-      String value = "";
-      String type = "";
-      String action = "";
-
-      Activity currentActivity = getCurrentActivity();
-
-      if (currentActivity != null) {
-        Intent intent = currentActivity.getIntent();
-        action = intent.getAction();
-        type = intent.getType();
-        if (type == null) {
-          type = "";
-        }
-        if (Intent.ACTION_SEND.equals(action) && "text/plain".equals(type)) {
-          value = intent.getStringExtra(Intent.EXTRA_TEXT);
+    public ShareModule(ReactApplicationContext reactContext) {
+        super(reactContext);
+    }
+
+    @Override
+    public String getName() {
+        return "ReactNativeShareExtension";
+    }
+
+    @ReactMethod
+    public void close() {
+        getCurrentActivity().finish();
+    }
+
+    @ReactMethod
+    public void data(Promise promise) {
+        promise.resolve(processIntent());
+    }
+
+    public WritableArray processIntent() {
+        WritableArray dataArrayMap = Arguments.createArray();
+        Set<String> mediaTypesSupported = new HashSet<String>();
+        mediaTypesSupported.add("video");
+        mediaTypesSupported.add("audio");
```

Comment:
@Karanbsankey what is audio used for? We don't support adding pure audio file. Is this needed for sharing a video with sound?